src/Entities/Project.php

```diff
@@ -37,6 +57,27 @@ public function getIsOverdue(): ?bool
         return $this->isOverdue;
     }
 
+    public function toAssociativeArrayAllProperties(): array
+    {   
+        $usersArray = $this->users;
+
+        foreach($usersArray as $user){
+            //this method comes from User class
+            $users[] = $user->toAssociativeArray();
```

Comment:
`users []` needs to be declared before it is first used, at the top of the function?